hw04_lru_cache/list.go

```diff
@@ -17,8 +17,87 @@ type ListItem struct {
 }
 
 type list struct {
-	List // Remove me after realization.
-	// Place your code here.
+	size int
+	head *ListItem
+	tail *ListItem
+}
+
+func (l list) Len() int {
+	return l.size
+}
+
+func (l list) Front() *ListItem {
+	return l.head
+}
+
+func (l list) Back() *ListItem {
```

Comment:
```suggestion
func (l *list) Back() *ListItem {
```
Гошные гайдлайны не рекомендуют миксовать value и pointer ресиверы в одном интерфейсе. Тут либо так либо так. Но в основном все пользуются пойнтерами.